tools/DatasetExtraction/dataset.py

```diff
@@ -0,0 +1,37 @@
+from abc import ABCMeta, abstractmethod
+from accessify import private, protected
+import chord
+
+class Dataset:
+    #i - song, j - chord
+    songlist = [[]]
+
+    @protected
+    def validate(self):
+        for song in self.songlist:
+            for ch in song:
+                if not isinstance(ch, chord):
+                    raise Exception("Incorrect chord type")
+                    #will also chech cq_max

```

Comment:
this is not what we need from this validation, really. And it should be abstract. For now forget about it and just create an abstract stub.